packages/block-editor/src/components/list-view/link-ui.js

```diff
@@ -0,0 +1,167 @@
+// Note: this file is copied directly from packages/block-library/src/navigation-link/link-ui.js
```

Comment:
Should this file be hooked up via the navigation block instead? I don't have enough context, but it feels like if this is only for the navigation that it should be within the navigation instead.

Reply:
I agree, I think it could be preferable to keep this sort of logic in the block, to try to avoid adding too much extra complexity to the `ListView` component itself, if possible. I've added a similar comment in https://github.com/WordPress/gutenberg/pull/49417#discussion_r1189275377.

Overall, I think it'd be good if we can avoid the list view having to know too much about particular blocks.